nautobot/core/tests/test_filters.py

```diff
@@ -87,26 +87,62 @@ def setUp(self):
             location_type=self.location_type,
             status=status,
         )
+        """
+        To ensure child objects with parents of the same name are not included in the results for filtering
+        we create parents with the same name under different ancestors.
+        """
+        self.parent_location_same_name_1 = dcim_models.Location.objects.create(
+            parent=self.child_location_1,
+            name="Test Parent Location Same Name",
+            location_type=self.location_type,
+            status=status,
+        )
+        self.parent_location_same_name_2 = dcim_models.Location.objects.create(
+            parent=self.child_location_2ab,
+            name="Test Parent Location Same Name",
+            location_type=self.location_type,
+            status=status,
+        )
+        self.child_location_same_name_1 = dcim_models.Location.objects.create(
+            parent=self.parent_location_same_name_1,
+            name="Test Child Location Same Name Parent 1",
+            location_type=self.location_type,
+            status=status,
+        )
+        self.child_location_same_name_2 = dcim_models.Location.objects.create(
+            parent=self.parent_location_same_name_2,
+            name="Test Child Location Same Name Parent 2",
+            location_type=self.location_type,
+            status=status,
+        )
         self.queryset = dcim_models.Location.objects.filter(name__icontains="Test Child Location")
 
     def test_filter_single_name(self):
         kwargs = {"parent": [self.parent_location_1.name]}
         qs = self.LocationFilterSet(kwargs, self.queryset).qs
 
-        self.assertQuerysetEqual(qs, [self.child_location_1])
+        self.assertQuerysetEqual(qs, [self.child_location_1, self.child_location_same_name_1])
```

Comment:
Hmm, something seems off here - why isn't `self.parent_location_same_name_1` expected to be part of this result too? It's a child of `child_location_1` and a parent of `child_location_same_name_1` so why are they both returned but not it?

Reply:
Note the name filter of the base QuerySet.

Reply:
This line 90/118 filters out all the locations with name "Test Parent Location" etc.